beesdoo_shift_swap/models/dated_template.py

```diff
@@ -0,0 +1,175 @@
+        first_shift = list_shift[0]
+        last_template = first_shift.task_template_id
+        new_template = first_shift.task_template_id
+        last_date = first_shift.start_time
+        new_date = first_shift.start_time
+
+        first_timeslot = self.env["beesdoo.shift.template.dated"].new()
+        first_timeslot.template_id = first_shift.task_template_id
+        first_timeslot.date = first_shift.start_time
+        timeslot_rec |= first_timeslot
+
+        shift_generated_list = []
+        for shift_rec in list_shift:
+            shift_generated_list.append(shift_rec)
+
+        for i in range(1, len(shift_generated_list)):
+            if last_template != new_template or last_date != new_date:
+                timeslot = self.env["beesdoo.shift.template.dated"].new()
+                timeslot.template_id = shift_generated_list[i-1].task_template_id
+                timeslot.date = shift_generated_list[i-1].start_time
+                timeslot_rec |= timeslot
+                new_template = shift_generated_list[i-1].task_template_id
+                new_date = shift_generated_list[i-1].start_time
+            last_template = shift_generated_list[i].task_template_id
+            last_date = shift_generated_list[i].start_time
+
+        shift_generated_list.clear()
+
+        return timeslot_rec
+
+    @api.model
+    def display_timeslot(self,my_timeslot):
+
+        start_date = datetime.now()
+
+        #generate timeslot of the shift already generated
+
+        shift_generated = (self.env["beesdoo.shift.shift"].sudo().search([
+            ("start_time", ">", start_date.strftime("%Y-%m-%d %H:%M:%S"))
+        ],
+        order="start_time, task_template_id, task_type_id"))
+
+        timeslot_rec = self.swap_shift_to_timeslot(shift_generated)
+
+        # generate timeslot of the shift not generated
+        last_sequence = int(
+            self.env["ir.config_parameter"]
+                .sudo()
+                .get_param("last_planning_seq")
+        )
+        next_planning = self.env["beesdoo.shift.planning"]._get_next_planning(last_sequence)
+        next_planning_date = fields.Datetime.from_string(
+            self.env["ir.config_parameter"]
+                .sudo()
+                .get_param("next_planning_date", 0)
+        )
+        #TODO : create system parameters for end_date
+        next_swap_limit = int(
+            self.env["ir.config_parameter"]
+                .sudo()
+                .get_param("beesdoo_shift.day_limit_swap")
+        )
+        end_date = my_timeslot.date + timedelta(days=next_swap_limit)
+        next_planning = next_planning.with_context(visualize_date=next_planning_date)
+        shift_recset = self.env["beesdoo.shift.shift"]
+
+        '''
+        if not next_planning.task_template_ids:
+            _logger.error(
+                "Could not generate next planning: no task template defined."
+            )
+            return
+        '''
```

Comment:
You should leave it uncommented or remove it. :)

Reply:
import longging module